cpp/src/bitmask/null_mask.cu

```diff
@@ -150,6 +150,27 @@ __global__ void count_set_bits_kernel(bitmask_type const *bitmask,
   }
 }
 
+/**
+ * @brief Convenience function to get offset word from a bitmask
+ * 
+ * @see copy_offset_bitmask
+ * @see offset_bitmask_and
+ */
+__device__ bitmask_type get_mask_offset_word(
+  bitmask_type const *__restrict__ source,
+  size_type _word_index,
+  size_type bit_offset,
+  size_type number_of_mask_words)
+{
+  size_type source_word_index = _word_index + word_index(bit_offset);
+  bitmask_type curr_word = source[source_word_index];
+  bitmask_type next_word = 0;
+  if (_word_index + 1 < number_of_mask_words) {
```

Comment:
Note that copy_offset_bitmask is updated in the main branch.

https://github.com/rapidsai/cudf/blob/branch-0.11/cpp/src/bitmask/null_mask.cu#L172

Also note that this check is inaccurate.

Say bit_offset is 8, _word_index is 0 and numbrer_of_mask_words is 1.

You still need to read two words from source to get a word for destination.

Reply:
I merged the mainline changes into the `get_mask_offset_word` function here: https://github.com/rapidsai/cudf/pull/3507/files#diff-ff4d421fc7fc9b9fc8ae929424491a8eR193. Let me know (in PR #3507) if it doesn't look correct.